hooks/useNodeReleasesDataApi.ts

```diff
@@ -0,0 +1,86 @@
+import { useRouter } from 'next/router';
+import { useMemo } from 'react';
+import useSWR from 'swr';
+import type { NodeReleaseData } from '../types';
+
+const fetcher = (...args: Parameters<typeof fetch>) =>
+  fetch(...args).then(res => res.json());
+
+const getNodeReleaseStatus = (
+  now: Date,
+  currentStart?: string,
+  ltsStart?: string,
+  maintenanceStart?: string,
+  endOfLife?: string
+) => {
+  if (endOfLife) {
+    const endOfLifeDate = new Date(endOfLife);
+    if (now > endOfLifeDate) {
+      return 'End-of-life';
+    }
+  }
+
+  if (maintenanceStart) {
+    const maintenanceStartDate = new Date(maintenanceStart);
+    if (now > maintenanceStartDate) {
+      return 'Maintenance LTS';
+    }
+  }
+
+  if (ltsStart) {
+    const ltsStartDate = new Date(ltsStart);
+    if (now > ltsStartDate) {
+      return 'Active LTS';
+    }
+  }
+
+  if (currentStart) {
+    const currentStartDate = new Date(currentStart);
+    if (now >= currentStartDate) {
+      return 'Current';
+    }
+  }
+
+  return 'Pending';
+};
+
+export const useNodeReleasesDataApi = (): NodeReleaseData[] => {
+  const { basePath } = useRouter();
+
+  const { data = [] } = useSWR<any[]>(
+    `${basePath}/static/node-releases-data.json`,
+    fetcher
+  );
+
+  const nodeReleasesData = useMemo(() => {
+    const now = new Date();
+
+    return data.map(raw => {
+      const status = getNodeReleaseStatus(
+        now,
```

Comment:
Since you need the data only here: I instead do `Date.now()` inline here

Reply:
It is inside `.map()`, inline `new Date()` will cause it to be called many times.
I think it's better to stick to consistent `now` value.

Reply:
True, didn't notice that.